extensions/libraries/redcore/form/fields/rdatepicker.php

```diff
@@ -61,6 +166,54 @@ protected function getInput()
 		// We will add a rdatepicker class to solve common styling issues
 		$this->element['class'] = $this->element['class'] ? 'rdatepicker ' . $this->element['class'] : 'rdatepicker';
 
+		// Get some system objects.
+		$config = JFactory::getConfig();
+		$user = JFactory::getUser();
+		$phpFormat = $this->dateFormatJQueryUIToPHP($this->getAttribute('dateFormat'));
+
+		if (isset($this->element['showTime']) && $this->element['showTime'] == 'true')
+		{
+			$phpFormat .= ' H:i:s';
```

Comment:
This is where you start to see that mixing two concepts is a wrong idea. I wouldn't add a hardcoded time format here. Also you already have a function to retrieve the format (`dateFormatJQueryUIToPHP`). Why process things again outside it?